packages/datadog-plugin-child_process/test/index.spec.js

```diff
@@ -0,0 +1,457 @@
+    args.push(command.join(' '))
+  } else {
+    args.push(command[0], command.slice(1))
+  }
+
+  args.push(options)
+
+  return args
+}
+
+describe('Child process plugin', () => {
+  describe('unit tests', () => {
+    let tracerStub, configStub, spanStub
+
+    beforeEach(() => {
+      spanStub = {
+        setTag: sinon.stub(),
+        finish: sinon.stub()
+      }
+
+      tracerStub = {
+        startSpan: sinon.stub()
+      }
+    })
+
+    afterEach(() => {
+      sinon.restore()
+    })
+
+    it('should call startSpan with proper parameters', () => {
+      const shellPlugin = new ChildProcessPlugin(tracerStub, configStub)
+
+      shellPlugin.start({ command: 'ls -l' })
+
+      expect(tracerStub.startSpan).to.have.been.calledOnceWithExactly(
+        'command_execution',
+        {
+          childOf: undefined,
+          tags: {
+            component: 'subprocess',
+            'service.name': undefined,
+            'resource.name': 'ls',
+            'span.kind': undefined,
+            'span.type': 'system',
+            'cmd.exec': JSON.stringify([ 'ls', '-l' ])
+          },
+          integrationName: 'system'
+        }
+      )
+    })
+
+    it('should call startSpan with cmd.shell property', () => {
+      const shellPlugin = new ChildProcessPlugin(tracerStub, configStub)
+
+      shellPlugin.start({ command: 'ls -l', shell: true })
+
+      expect(tracerStub.startSpan).to.have.been.calledOnceWithExactly(
+        'command_execution',
+        {
+          childOf: undefined,
+          tags: {
+            component: 'subprocess',
+            'service.name': undefined,
+            'resource.name': 'ls',
+            'span.kind': undefined,
+            'span.type': 'system',
+            'cmd.shell': JSON.stringify([ 'ls', '-l' ])
+          },
+          integrationName: 'system'
+        }
+      )
+    })
+
+    it('should truncate path and blank last argument', () => {
+      const shellPlugin = new ChildProcessPlugin(tracerStub, configStub)
+      const path = '/home/'.padEnd(4000 * 8, '/')
+      const command = 'ls -l' + ' ' + path + ' -t'
+
+      shellPlugin.start({ command, shell: true })
+
+      expect(tracerStub.startSpan).to.have.been.calledOnceWithExactly(
+        'command_execution',
+        {
+          childOf: undefined,
+          tags: {
+            component: 'subprocess',
+            'service.name': undefined,
+            'resource.name': 'ls',
+            'span.kind': undefined,
+            'span.type': 'system',
+            'cmd.shell': JSON.stringify([ 'ls', '-l', '/h', '' ])
+          },
+          integrationName: 'system'
+        }
+      )
+    })
+
+    it('should truncate first argument and blank the rest', () => {
+      const shellPlugin = new ChildProcessPlugin(tracerStub, configStub)
+      const option = '-l'.padEnd(4000 * 8, 't')
+      const path = '/home'
+      const command = 'ls' + ' ' + option + ' ' + path + ' -t'
```

Comment:
way more readable if you ask me
```suggestion
      const command = `ls ${option} ${path} -t`
```

Reply:
makes sense